### PR TITLE
Wip/hughsie/precommit quirks

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — DISMISSED** [review]

File: contrib/ci/check-quirks.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2021 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import glob
+import os
```
Comment (superm1) [inline contrib/ci/check-quirks.py:8]: unneeded

File: contrib/ci/check-quirks.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                        rc = 1
+                        continue
+                else:
+                    sections = line.split(" = ")
+                    if len(sections) != 2:
+                        print("{} has invalid line: {}".format(fn, line))
+                        rc = 1
+                        continue
+                    for section in sections:
+                        if section.startswith(" ") or section.endswith(" "):
```
Comment (superm1) [inline contrib/ci/check-quirks.py:37]: ideally is there a way to detect any whitespace instead of just a space?

File: contrib/ci/check-quirks.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+
+def test_files() -> int:
+
+    rc: int = 0
+
+    for fn in glob.glob("**/*.quirk", recursive=True):
+        with open(fn, "r") as f:
+            for line in f.read().split("\n"):
+                if line.startswith(" ") or line.endswith(" "):
```
Comment (superm1) [inline contrib/ci/check-quirks.py:19]: can we detect the deprecated `DeviceInstanceId` prefix too?

File: contrib/ci/check-quirks.py (hunk 1 above)
Comment (hughsie) [reply contrib/ci/check-quirks.py:8]: Fixed.

File: contrib/ci/check-quirks.py (hunk 3 above)
Comment (hughsie) [reply contrib/ci/check-quirks.py:19]: Yup, good idea. And Guid.

File: contrib/ci/check-quirks.py (hunk 2 above)
Comment (hughsie) [reply contrib/ci/check-quirks.py]: Sure, is, fixed.

File: contrib/ci/check-quirks.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                        print("{} has invalid section header: {}".format(fn, line))
+                        rc = 1
+                        continue
+                    for deprecated in ["DeviceInstanceId", "Guid"]:
+                        if line.find(deprecated) != -1:
+                            print("{} has deprecated prefix: {}".format(fn, deprecated))
+                            rc = 1
+                            continue
+                else:
+                    sections = line.split(" = ")
```
Comment (superm1) [inline contrib/ci/check-quirks.py:35]: what about ` =` or `= `?  I think need to also check those types of things

Comment (hughsie) [reply contrib/ci/check-quirks.py:35]: then we'd fail the `len(sections) != 2` check the line below.

**Review by superm1 — APPROVED** [review]

**Pull request merged**